geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateDefinedIndexesCommand.java

```diff
@@ -89,23 +88,25 @@ public Result createDefinedIndexes(
 
           if (cliFunctionResult.isSuccessful()) {
             successfulMembers.add(cliFunctionResult.getMemberIdOrName());
-
-            if (xmlEntity.get() == null) {
-              xmlEntity.set(cliFunctionResult.getXmlEntity());
-            }
+            xmlEntities.add(cliFunctionResult.getXmlEntity());
           } else {
             final String exceptionMessage = cliFunctionResult.getMessage();
             Set<String> failedMembers = indexOpFailMap.get(exceptionMessage);
 
             if (failedMembers == null) {
               failedMembers = new TreeSet<>();
             }
+
             failedMembers.add(cliFunctionResult.getMemberIdOrName());
             indexOpFailMap.put(exceptionMessage, failedMembers);
           }
         }
       }
 
+      // TODO:
+      // with the current logic some indexes might be correctly created in some members and
+      // fail in others; but the user is never notified about these failures if there's at least one
+      // successful creation...
```

Comment:
I'm not sure how consistent we are on this, but I think the command should fail if there are **any** errors at all. I know that we do this in `CreateRegionCommand`. We're adding options to allow for idempotency in all commands, so cleaning up from a partial failure will become very easy. i.e. on failure you can then issue: `destroy index --if-exists`

Reply:
I'm not 100% sure about this one... currently creation can be successful on some members and fail on others, in that case we should report to the user which indexes were successfully created and which ones failed, adding the details about the members on which the operation failed/succeeded. I've created GEODE-3916 to address this.
Regarding this pull request, I'll add more details in the comment for future reference.